File: src/main/java/red/tetracube/slurpanize/baker/authentication/payloads/AuthenticationRequest.java
package red.tetracube.slurpanize.baker.authentication.payloads;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

public class AuthenticationRequest {

    @NotNull
    @NotEmpty
    @JsonProperty("username")
    public String username;

    @NotNull
    @NotEmpty
    @JsonProperty("password")
    public String password;

}

File: src/main/kotlin/tetracubered/slurpanizeBaker/authentication/payloads/AuthenticationRequest.kt
package tetracubered.slurpanizeBaker.authentication.payloads

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

data class AuthenticationRequest @JsonCreator constructor(
    @field:NotNull
    @field:NotEmpty
    @JsonProperty("username")
    val username: String? = null,

    @field:NotNull
    @field:NotEmpty
    @JsonProperty("password")
    val password: String? = null
)